indexer/src/main/java/au/org/aodn/esindexer/service/VocabServiceImpl.java

```diff
@@ -88,55 +88,58 @@ this method for analysing the vocabularies of a record aka bottom-level vocabs (
      */
     public List<String> extractVocabLabelsFromThemes(List<ThemesModel> themes, String vocabType) throws IOException {
         List<String> results = new ArrayList<>();
-        // Iterate over the top-level vocabularies
-        List<JsonNode> vocabs = switch (vocabType) {
-            case AppConstants.AODN_DISCOVERY_PARAMETER_VOCABS -> self.getParameterVocabs();
-            case AppConstants.AODN_PLATFORM_VOCABS -> self.getPlatformVocabs();
-            case AppConstants.AODN_ORGANISATION_VOCABS -> self.getOrganisationVocabs();
-            default -> new ArrayList<>();
-        };
+        if (vocabType.equals(AppConstants.AODN_ORGANISATION_VOCABS)) {
+            // TODO: the logics for mapping record's organisation vocabs are heavily customised for a manual approach, AI now or later?
```

Comment:
should I do it the old school if-else, or leave here empty for Yuxuan's future magic? @utas-raymondng

Reply:
How complicate the logic will be?